contrib/in-memory-accumulo/src/main/java/datawave/accumulo/inmemory/InMemoryBatchScanner.java

```diff
@@ -69,6 +88,27 @@ public void setRanges(Collection<Range> ranges) {
         }
         return chain;
     }
+
+    @Override
+    public Iterator<Entry<Key,Value>> rebuild(Key lastKey) {
+        // rebuild the ranges
+        List<Range> newRanges = new ArrayList<>();
+        boolean found = false;
+        for (Range range : ranges) {
+            if (found) {
+                newRanges.add(range);
+            } else if (range.contains(lastKey)) {
```

Comment:
Won't this potentially remove ranges? If there were, say 3 ranges. The first does not contain lastKey, the second did, and the third did not. For the first range, found would be false and also range.contains(lastKey) is false, so the range wouldn't be added to the newRanges set.

Reply:
In the IteratorChain used in the previous method, the ranges are processed in order.  So the idea is to subset the ranges starting with the first range that contains the lastKey (which was the last key returned by the iterator).  All ranges up to the range containing the lastKey need to be dropped, and the the range containing the key is reset to start with lastKey, and then all subsequent ranges are readded to the list.  I will add documentation to state the intention here.

Reply:
It needs to be more than processing the ranges in order, doesn't it? The ranges have to be sorted (and not overlapping at all). Do we have that guarantee?

Reply:
That is a very astute observation.  However sorting is not required, only non-overlapping.  Since the ranges are processed in the same order in the list, I only need to find the first one containing the key returned.  In terms of the non-overlapping assumption, the cost of not taking that into account would be duplicate keys being returned.  For the sake of the test cases, I do not believe this needs to be taken into account.  However to be thorough, I probably should.  This will take me a little longer to do.

Reply:
Good point. I think I was thinking of threading going on. Anyway, never mind about the sorting. For overlapping ranges, if the real batch scanner merges overlapping ranges then I think we should do one of the following:
1. Merge overlapping ranges here too.
2. Detect that there are overlapping ranges and throw an exception so the test case will at least fail
If merging looks like it's going to be a lot of work, then I would do the second since that should be trivial. If we're actually sending overlapping ranges in any test case, I'd rather detect it and have test cases fail then get duplicate keys back.

Reply:
I ensure non-overlapping in the setRanges method above.  Done.